pkg/reconcile/reconcilers.go

```diff
@@ -185,49 +190,42 @@ func (r *defaultReconciler) ReconcileSchema(ctx context.Context, req ReaperReque
 	}
 }
 
-func (r *defaultReconciler) checkForCassandraDatacenterReadiness(ctx context.Context, req ReaperRequest) error {
+func (r *defaultReconciler) checkForCassandraDatacenterReadiness(ctx context.Context, req ReaperRequest) (bool, error) {
 	reaper := req.Reaper
-	cassandra := *reaper.Spec.ServerConfig.CassandraBackend
-	cassdc := &cassdcapi.CassandraDatacenter{}
+	cassdc, err := r.cassandraDatacenter(ctx, reaper)
+	//if meta.IsNoMatchError(err) {
+	//	return true, nil
+	//}
+	//
+	if err != nil {
+		req.Logger.Error(err, "failed to fetch CassandraDatacenter")
+		return false, err
+	}
 
-	fetchStatus := func() error {
-		cassdcKey := types.NamespacedName{Namespace: reaper.Namespace, Name: cassandra.DatacenterName()}
-		err := r.Client.Get(ctx, cassdcKey, cassdc)
-		if runtime.IsNotRegisteredError(err) {
-			return nil
-		}
-		if err != nil {
-			req.Logger.Error(err, "failed to fetch CassandraDatacenter")
-			return err
+	if value, found := cassdc.Annotations["reaper.cassandra-reaper.io/instance"]; found {
+		if value == reaper.Name {
+			if isCassdcReady(cassdc) {
+				return true, nil
+			}
+			return false, nil
 		}
-		return nil
 	}
 
-	maxWaitTime := time.Now().Add(30 * time.Second)
+	return true, nil
+}
 
-	for {
-		err := fetchStatus()
-		if err != nil {
-			return err
-		}
-		if cassdc.Name != cassandra.DatacenterName() {
-			// Not running under k8ssandra or this object was empty (it will error elsewhere)
-			return nil
-		}
-		if value, found := cassdc.Annotations["reaper.cassandra-reaper.io/instance"]; found {
-			if value == reaper.Name {
-				if isCassdcReady(cassdc) {
-					return nil
-				}
-			}
-		}
-		// Sleep half a second and then try again
-		time.Sleep(500 * time.Millisecond)
-		if time.Now().After(maxWaitTime) {
-			// This will cause a CrashLoopBackoff, but it will be retried later - however a good indication of perf issue
-			return fmt.Errorf("CassandraDatacenter did not start")
-		}
+func (r *defaultReconciler) cassandraDatacenter(ctx context.Context, reaper *api.Reaper) (*cassdcapi.CassandraDatacenter, error) {
+	cassandra := *reaper.Spec.ServerConfig.CassandraBackend
+	cassdc := &cassdcapi.CassandraDatacenter{}
+
+	targetDc := cassandra.CassandraDatacenter
+	if targetDc.Namespace == "" {
+		targetDc.Namespace = reaper.Namespace
 	}
+
+	cassdcKey := types.NamespacedName{Namespace: targetDc.Namespace, Name: targetDc.Name}
+	_ = r.Client.Get(ctx, cassdcKey, cassdc)
+	return cassdc, nil
```

Comment:
Shouldn't we also return the error returned from the `r.Client.Get(ctx, cassdcKey, cassdc)` call?

Reply:
Problem is that it doesn't work in the tests (it will always throw an error).